lib/ash_authentication/token_resource/transformer.ex

```diff
@@ -156,11 +156,73 @@ defmodule AshAuthentication.TokenResource.Transformer do
              get_token_action_name,
              &build_get_token_action(&1, get_token_action_name)
            ),
-         :ok <- validate_get_token_action(dsl_state, get_token_action_name) do
+         :ok <- validate_get_token_action(dsl_state, get_token_action_name),
+         :ok <- validate_primary_key(dsl_state) do
       {:ok, dsl_state}
     end
   end
 
+  defp validate_primary_key(dsl_state) do
+    dsl_state
+    |> Ash.Resource.Info.attributes()
+    |> Enum.filter(& &1.primary_key?)
+    |> Enum.map(& &1.name)
```

Comment:
```suggestion
    |> Ash.Resource.Info.primary_key()
```

Reply:
We can't use this yet because this transformer runs before the id transformer.